src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Utilities/SpeculationAnalyzer.cs

```diff
@@ -722,28 +722,65 @@ private bool ReplacementBreaksQueryClause(QueryClauseSyntax originalClause, Quer
                 !SymbolInfosAreCompatible(originalClauseInfo.OperationInfo, newClauseInfo.OperationInfo);
         }
 
-        protected override bool ReplacementIntroducesErrorType(ExpressionSyntax originalExpression, ExpressionSyntax newExpression)
+        protected override bool ReplacementIntroducesDisallowedNullType(
```

Comment:
broke this into two checks.  a check for error-types (always bad), and a check for a null type (sometimes bad, sometimes ok).